metakb/service_schema.py

```diff
@@ -0,0 +1,259 @@
+                         model: Type['NestedStatementResponse']) -> None:
+            """Configure OpenAPI schema"""
+            if 'title' in schema.keys():
+                schema.pop('title', None)
+            for prop in schema.get('properties', {}).values():
+                prop.pop('title', None)
+            # TODO: Update
+            schema['example'] = {}
+
+
+class SearchQuery(BaseModel):
+    """Queries for the Search Endpoint."""
+
+    variation: Optional[StrictStr]
+    disease: Optional[StrictStr]
+    therapeutic: Optional[StrictStr]
+    gene: Optional[StrictStr]
+    statement_id: Optional[StrictStr]
+    detail: StrictBool
+
+    class Config:
+        """Configure examples."""
+
+        @staticmethod
+        def schema_extra(schema: Dict[str, Any],
+                         model: Type['SearchQuery']) -> None:
+            """Configure OpenAPI schema"""
+            if 'title' in schema.keys():
+                schema.pop('title', None)
+            for prop in schema.get('properties', {}).values():
+                prop.pop('title', None)
+            schema['example'] = {
+                "variation": "NP_005219.2:p.Leu858Arg",
+                "disease": "Lung Non-small Cell Carcinoma",
+                "therapeutic": "Afatinib",
+                "statement_id": "civic.eid:2997",
+                "detail": False
+            }
+
+
+class SearchStatementsQuery(BaseModel):
+    """Queries for the Search Endpoint."""
+
+    variation: Optional[StrictStr]
+    disease: Optional[StrictStr]
+    therapy: Optional[StrictStr]
+    gene: Optional[StrictStr]
+    statement_id: Optional[StrictStr]
+
+    class Config:
+        """Configure examples."""
+
+        @staticmethod
+        def schema_extra(schema: Dict[str, Any],
+                         model: Type['SearchStatementsQuery']) -> None:
+            """Configure OpenAPI schema"""
+            if 'title' in schema.keys():
+                schema.pop('title', None)
+            for prop in schema.get('properties', {}).values():
+                prop.pop('title', None)
+            schema['example'] = {
+                "variation": "NP_005219.2:p.Leu858Arg",
+                "disease": "Lung Non-small Cell Carcinoma",
+                "therapeutic": "Afatinib",
+                "statement_id": "civic.eid:2997"
+            }
+
+
+class Matches(BaseModel):
+    """Statements and Propositions that match the queried parameters."""
+
+    statements: List[StrictStr]
+    propositions: List[StrictStr]
+
+    class Config:
+        """Configure examples."""
+
+        @staticmethod
+        def schema_extra(schema: Dict[str, Any],
+                         model: Type['Matches']) -> None:
+            """Configure OpenAPI schema"""
+            if 'title' in schema.keys():
+                schema.pop('title', None)
+            for prop in schema.get('properties', {}).values():
+                prop.pop('title', None)
+            schema['example'] = {
```

Comment:
fyi, there are some examples of single quotes on this page even though it looks like you went through and double-quoted everything elsewhere (maybe double check that a linter is active?)

Reply:
This commit was probably done before I added the flake8 pre-commit hook. It will also be addressed in #193